### PR TITLE
fix(curriculum): add clarification to question and escape backslash

### LINKED ISSUE #38928: Python for Everybody - Regular Expressions
**Describe your problem and how to reproduce it:**
I think the wording of the question should be re-done. Whilst I perfectly understood what was being asked for, there are **3** correct answers, as `.` does match _"a white space character"_.

**Add a Link to the page with the problem:**
https://www.freecodecamp.dev/learn/scientific-computing-with-python/python-for-everybody/regular-expressions

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38928

Simple fix to question and answers

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.english.md [hunk 1]
```diff
@@ -17,13 +17,13 @@ videoId: Yud_COr6pZo
 
 ```yml
 question:
-  text: 'Which regex matches a white space character?'
+  text: 'Which regex only matches a white space character?'
```
Comment (moT01) [inline curriculum/challenges/english/07-scientific-computing-with-python/python-for-everybody/regular-expressions.english.md:20]:
```suggestion
  text: 'Which regex matches only a white space character?'
```

**Review by moT01 — COMMENTED** [review]
It sounded a little funny like that, what do you think of my suggestion @Sky020?

**Review by moT01 — DISMISSED** [review]
LGTM 🎉

**Review by erictleung — DISMISSED** [review]
LGTM!

**Comment by erictleung** [comment]
Looks like there are some merge conflicts that appear to have made this change already, but with some different syntax (the green additions are from master right now)

```diff
-  text: 'Which regex matches only a white space character?'
+  text: |
+    Which regex only matches a white space character?
  answers:
-    - '\S'
+    - |
+        \S
-    - '\s'
+    - |
+        \s
-    - '.'
+    - |
+        .
-    - '\_'
+    - |
+        \_
-    - '\\.'
+   - |
+        \\.
```

**Comment by Sky020** [comment]
> I prefer the new changes from this PR as opposed to the current Markdown.

@erictleung , I kept the MD, because every other file uses it, and I think the consistency is best.

**Comment by ojeytonwilliams** [comment]
The more verbose syntax currently on master is deliberate, but there might be a better way.

So, the reasoning goes like this. By using the syntax 

```yaml
- |
  markdown
```

the user can write whatever they want, the only thing that matters is the indentation.  However, if you skip the pipe then the following will not compile

````yaml
  answers:
    - *syntax* error
````

while 

````yaml
  answers:
    - plain text
````

will, so you have to use

````yaml
  answers:
    - '*syntax* error'
````

and if you want single quotes you have to do things like this

````yaml
  answers:
    - '''*syntax* error'''
````

or 

````yaml
  answers:
    - "'*syntax* error'"
````

which is messy.

**Review by erictleung — APPROVED** [review]

**Comment by erictleung** [comment]
@ojeytonwilliams oh thanks for the clarification. That does seem messy... 😅

**Comment by scissorsneedfoodtoo** [comment]
Thank you for submitting a PR to fix this @Sky020, and everyone else for your review/input. @ojeytonwilliams, had no idea how many hoops you have to jump through to get single quotes!

**Comment by ojeytonwilliams** [comment]
Indeed, yaml isn't always the easiest thing to work with.  I think it was some of @hassaanp's contributions where I saw you could simplify things with the |, so, thanks Hassaan!

**Pull request merged**